### PR TITLE
Result service butterfly bumblebee census 170074940

### PR DESCRIPTION
**Pull request by Hvar83**

### CODE REVIEW

**Comment by olzraiti** [comment]
Is this ready for reviewing or still under development? I took a quick look and noticed for example that this breaks the result service for NAFI (https://170074940.dev.laji.fi/project/MHL.6/stats has a "routes" sub nav link which doesn't do anything and there reads "content" after the title).

**Comment by Hvar83** [comment]
@olzraiti As i still don't receive anything about resultServiceType from API, i set in project-form component form resultService to nafiBumblebee. Just remove that line and nafi is working

**Comment by olzraiti** [comment]
The forms (MHL.59 & MHL.89) didn't have any result service set. I added the result service for them just now.

**Comment by Hvar83** [comment]
@olzraiti ok

**Comment by olzraiti** [comment]
Let's follow the new development process here, so that this enters into reviewing only after the task has been approved in Pivotal. Please notify here when it's been approved.

**Comment by MLumme** [comment]
@olzraiti this has now been approved for review bu PO.

**Review by olzraiti — CHANGES_REQUESTED** [review]
The comments have mostly very minor stuff (unused vars etc), but on top of that I also get warnings when I start the dev server:

```
Error: projects/laji/src/app/+project-form/results/nafi-bumblebee-result/nafi-bumblebee-result-filters/nafi-bumblebee-result-filters.component.html:21:48 - error TS2339: Property 'value' does not exist on type 'EventTarget'.

21           (change)="onDateChange($event.target.value !== '0' ? $event.target.value : undefined)"
                                                  ~~~~~

  projects/laji/src/app/+project-form/results/nafi-bumblebee-result/nafi-bumblebee-result-filters/nafi-bumblebee-result-filters.component.ts:9:16
    9   templateUrl: './nafi-bumblebee-result-filters.component.html',
                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    Error occurs in the template of component NafiBumblebeeResultFiltersComponent.


Error: projects/laji/src/app/+project-form/results/nafi-bumblebee-result/nafi-bumblebee-result-filters/nafi-bumblebee-result-filters.component.html:21:78 - error TS2339: Property 'value' does not exist on type 'EventTarget'.

21           (change)="onDateChange($event.target.value !== '0' ? $event.target.value : undefined)"
                                                                                ~~~~~

  projects/laji/src/app/+project-form/results/nafi-bumblebee-result/nafi-bumblebee-result-filters/nafi-bumblebee-result-filters.component.ts:9:16
    9   templateUrl: './nafi-bumblebee-result-filters.component.html',
                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    Error occurs in the template of component NafiBumblebeeResultFiltersComponent.


Error: projects/laji/src/app/+project-form/results/nafi-bumblebee-result/nafi-bumblebee-result-filters/nafi-bumblebee-result-filters.component.html:7:48 - error TS2339: Property 'value' does not exist on type 'EventTarget'.

7           (change)="onYearChange($event.target.value !== '0' ? $event.target.value : undefined)"
                                                 ~~~~~

  projects/laji/src/app/+project-form/results/nafi-bumblebee-result/nafi-bumblebee-result-filters/nafi-bumblebee-result-filters.component.ts:9:16
    9   templateUrl: './nafi-bumblebee-result-filters.component.html',
                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    Error occurs in the template of component NafiBumblebeeResultFiltersComponent.


Error: projects/laji/src/app/+project-form/results/nafi-bumblebee-result/nafi-bumblebee-result-filters/nafi-bumblebee-result-filters.component.html:7:78 - error TS2339: Property 'value' does not exist on type 'EventTarget'.

7           (change)="onYearChange($event.target.value !== '0' ? $event.target.value : undefined)"
                                                                               ~~~~~

  projects/laji/src/app/+project-form/results/nafi-bumblebee-result/nafi-bumblebee-result-filters/nafi-bumblebee-result-filters.component.ts:9:16
    9   templateUrl: './nafi-bumblebee-result-filters.component.html',
                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    Error occurs in the template of component NafiBumblebeeResultFiltersComponent.


Error: projects/laji/src/app/+project-form/results/nafi-bumblebee-result/nafi-bumblebee-route/nafi-bumblebee-route.component.html:29:6 - error TS2322: Type 'string' is not assignable to type 'SEASON'.

29     [season]="date"
        ~~~~~~

  projects/laji/src/app/+project-form/results/nafi-bumblebee-result/nafi-bumblebee-route/nafi-bumblebee-route.component.ts:12:16
    12   templateUrl: './nafi-bumblebee-route.component.html',
                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    Error occurs in the template of component NafiBumblebeeRouteComponent.


Error: projects/laji/src/app/+project-form/results/nafi-bumblebee-result/nafi-bumblebee-route/nafi-bumblebee-route.component.html:31:6 - error TS2322: Type 'number' is not assignable to type 'string'.

31     [year]="activeYear"
        ~~~~

  projects/laji/src/app/+project-form/results/nafi-bumblebee-result/nafi-bumblebee-route/nafi-bumblebee-route.component.ts:12:16
    12   templateUrl: './nafi-bumblebee-route.component.html',
                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    Error occurs in the template of component NafiBumblebeeRouteComponent.


Error: projects/laji/src/app/+project-form/results/nafi-bumblebee-result/nafi-bumblebee-routes/nafi-bumblebee-routes-list/nafi-bumblebee-routes-list.component.html:16:6 - error TS2322: Type 'string' is not assignable to type 'ColumnMode | "standard" | "flex" | "force"'.

16     [columnMode]="columnMode"
        ~~~~~~~~~~

  projects/laji/src/app/+project-form/results/nafi-bumblebee-result/nafi-bumblebee-routes/nafi-bumblebee-routes-list/nafi-bumblebee-routes-list.component.ts:10:16
    10   templateUrl: './nafi-bumblebee-routes-list.component.html',
                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    Error occurs in the template of component NafiBumblebeeRoutesListComponent.
```

Also, I think all the variables & files should be renamed (nafi bumblebee -> syke insect projects). Should be possible with `sed` & `rename` in unix. I can try the renaming if you could check those errors above first.

File: projects/laji/src/app/+project-form/results/syke-insect-result/syke-insect-result.component.ts [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+  ) {
+    const now = new Date();
+    this.currentYear = now.getFullYear();
+    this.currentMonth = now.getMonth() + 1;
+    this.year = now.getFullYear() - 1;
+  }
+
+  ngOnInit() {
+    this.collectionId = this.qName.transform(this.form.collectionID);
+    this.lang = this.translate.currentLang;
```
Comment (olzraiti) [inline projects/laji/src/app/+project-form/results/syke-insect-result/syke-insect-result.component.ts:65]: rm unused var (didn't notice this last round).

**Review by olzraiti — CHANGES_REQUESTED** [review]
Found one more unnecessary var. Also the conflicts need to be fixed.

**Review by olzraiti — APPROVED** [review]
:+1:

**Pull request merged**